### PR TITLE
Core: Integrate Ref `instantiate` where possible

### PR DESCRIPTION
**Pull request by Repiteo**
Followup to #88183, which expanded `Ref<T>::instantiate()` to accept constructor arguments. This PR is focused on converting existing Ref constructors to the shorthand syntax where applicable. Does not convert *every* instance, as some constructors happen on the same line a Ref is declared; converting these would require a static method, and is more suitable as a separate PR.

### CODE REVIEW

File: editor/plugins/light_occluder_2d_editor_plugin.cpp [hunk 1]
```diff
@@ -36,7 +36,7 @@
 Ref<OccluderPolygon2D> LightOccluder2DEditor::_ensure_occluder() const {
 	Ref<OccluderPolygon2D> occluder = node->get_occluder_polygon();
 	if (!occluder.is_valid()) {
```
Comment (AThousandShips) [inline editor/plugins/light_occluder_2d_editor_plugin.cpp:38]:
```suggestion
	if (occluder.is_null()) {
```
While we're here

File: editor/plugins/navigation_polygon_editor_plugin.cpp [hunk 2]
```diff
@@ -39,7 +39,7 @@
 Ref<NavigationPolygon> NavigationPolygonEditor::_ensure_navpoly() const {
 	Ref<NavigationPolygon> navpoly = node->get_navigation_polygon();
 	if (!navpoly.is_valid()) {
```
Comment (AThousandShips) [inline editor/plugins/navigation_polygon_editor_plugin.cpp:41]:
```suggestion
	if (navpoly.is_null()) {
```

File: modules/gridmap/grid_map.cpp [hunk 3]
```diff
@@ -802,7 +802,7 @@ void GridMap::_octant_enter_world(const OctantKey &p_key) {
 				g.navigation_debug_edge_connections_instance = RenderingServer::get_singleton()->instance_create();
 			}
 			if (!g.navigation_debug_edge_connections_mesh.is_valid()) {
```
Comment (AThousandShips) [inline modules/gridmap/grid_map.cpp:804]:
```suggestion
			if (g.navigation_debug_edge_connections_mesh.is_null()) {
```

File: modules/gridmap/grid_map.cpp [hunk 4]
```diff
@@ -1387,7 +1387,7 @@ void GridMap::_update_octant_navigation_debug_edge_connections_mesh(const Octant
 	}
 
 	if (!g.navigation_debug_edge_connections_mesh.is_valid()) {
```
Comment (AThousandShips) [inline modules/gridmap/grid_map.cpp:1389]:
```suggestion
	if (g.navigation_debug_edge_connections_mesh.is_null()) {
```

File: scene/3d/navigation_agent_3d.cpp [hunk 5]
```diff
@@ -1077,7 +1077,7 @@ void NavigationAgent3D::_update_debug_path() {
 	}
 
 	if (!debug_path_mesh.is_valid()) {
```
Comment (AThousandShips) [inline scene/3d/navigation_agent_3d.cpp:1079]:
```suggestion
	if (debug_path_mesh.is_null()) {
```

File: scene/3d/navigation_link_3d.cpp [hunk 6]
```diff
@@ -57,7 +57,7 @@ void NavigationLink3D::_update_debug_mesh() {
 	}
 
 	if (!debug_mesh.is_valid()) {
```
Comment (AThousandShips) [inline scene/3d/navigation_link_3d.cpp:59]:
```suggestion
	if (debug_mesh.is_null()) {
```

File: scene/3d/navigation_obstacle_3d.cpp [hunk 7]
```diff
@@ -382,7 +382,7 @@ void NavigationObstacle3D::_update_fake_agent_radius_debug() {
 		fake_agent_radius_debug_instance = RenderingServer::get_singleton()->instance_create();
 	}
 	if (!fake_agent_radius_debug_mesh.is_valid()) {
```
Comment (AThousandShips) [inline scene/3d/navigation_obstacle_3d.cpp:384]:
```suggestion
	if (fake_agent_radius_debug_mesh.is_null()) {
```

File: scene/3d/navigation_obstacle_3d.cpp [hunk 8]
```diff
@@ -480,7 +480,7 @@ void NavigationObstacle3D::_update_static_obstacle_debug() {
 		static_obstacle_debug_instance = RenderingServer::get_singleton()->instance_create();
 	}
 	if (!static_obstacle_debug_mesh.is_valid()) {
```
Comment (AThousandShips) [inline scene/3d/navigation_obstacle_3d.cpp:482]:
```suggestion
	if (static_obstacle_debug_mesh.is_null()) {
```

File: scene/3d/path_3d.cpp [hunk 9]
```diff
@@ -89,7 +89,7 @@ void Path3D::_update_debug_mesh() {
 	}
 
 	if (!debug_mesh.is_valid()) {
```
Comment (AThousandShips) [inline scene/3d/path_3d.cpp:91]:
```suggestion
	if (debug_mesh.is_null()) {
```

**Review by AThousandShips — APPROVED** [review]
Looks good and improves readability quite substantially

File: scene/resources/navigation_mesh.cpp [hunk 10]
```diff
@@ -363,7 +363,7 @@ Ref<ArrayMesh> NavigationMesh::get_debug_mesh() {
 	}
 
 	if (!debug_mesh.is_valid()) {
```
Comment (AThousandShips) [inline scene/resources/navigation_mesh.cpp:365]:
```suggestion
	if (debug_mesh.is_null()) {
```

File: scene/3d/navigation_region_3d.cpp [hunk 11]
```diff
@@ -493,7 +493,7 @@ void NavigationRegion3D::_update_debug_mesh() {
 	}
 
 	if (!debug_mesh.is_valid()) {
```
Comment (AThousandShips) [inline scene/3d/navigation_region_3d.cpp:495]:
```suggestion
	if (debug_mesh.is_null()) {
```

File: scene/3d/navigation_region_3d.cpp [hunk 12]
```diff
@@ -670,7 +670,7 @@ void NavigationRegion3D::_update_debug_edge_connections_mesh() {
 	}
 
 	if (!debug_edge_connections_mesh.is_valid()) {
```
Comment (AThousandShips) [inline scene/3d/navigation_region_3d.cpp:672]:
```suggestion
	if (debug_edge_connections_mesh.is_null()) {
```

**Comment by fire** [comment]
I'm curious, we seem to flip between memnew and instantiate. Which one do we pick for reviews?

**Comment by Repiteo** [comment]
Ideally, we'd want to use instantiate over memnew where possible. As mentioned in the OP, the only places where that *isn't* feasible is when it's the same line `Ref` is declared. That could be solved with a static method along the lines of:
```cpp
// old
Ref<SomeType> some_value = Ref<SomeType>(memnew(SomeType()));

// new
Ref<SomeType> some_value = Ref<SomeType>::construct();
```
Initially I wanted to tackle that in a separate PR, but if desired I could integrate it here instead

**Comment by Repiteo** [comment]
…Alternatively, if keeping within the initial scope of the PR is desired, there's this:
```cpp
// old
Ref<SomeType> some_value = Ref<SomeType>(memnew(SomeType()));

// new
Ref<SomeType> some_value;
some_value.instantiate();
```
I'd say this is still more readable than using `memnew`, even if it's arguably more verbose. There's already places in the repo utilizing this exact structure, so it wouldn't be unprecedented.

However, this wouldn't remove the need for a static constructor. To demonstrate, here's a scenario where `instantiate` would *hurt* readability:
```cpp
// old
some_function(Ref<SomeType>(memnew(SomeType)));

// new (static)
some_function(Ref<SomeType>::construct());

// new (local)
Ref<SomeType> some_value;
some_value.instantiate();
some_function(some_value);
```
With all this in mind, I'll probably leave this concept alone for now. Maaaybe I'll sweep for cases that aren't reliant on one-liners, but I'd prefer to keep this PR as simple as possible

**Review by adamscott — APPROVED** [review]
Everything seems alright!

**Comment by adamscott** [comment]
> …Alternatively, if keeping within the initial scope of the PR is desired, there's this:
> 
> ```c++
> // old
> Ref<SomeType> some_value = Ref<SomeType>(memnew(SomeType()));
> 
> // new
> Ref<SomeType> some_value;
> some_value.instantiate();
> ```
> 
> I'd say this is still more readable than using `memnew`, even if it's arguably more verbose. There's already places in the repo utilizing this exact structure, so it wouldn't be unprecedented.

I'm all for the new behavior. I'm already coding in that way and it does what it says.

> However, this wouldn't remove the need for a static constructor. To demonstrate, here's a scenario where `instantiate` would _hurt_ readability:
> 
> ```c++
> // old
> some_function(Ref<SomeType>(memnew(SomeType)));
> 
> // new (static)
> some_function(Ref<SomeType>::construct());
> 
> // new (local)
> Ref<SomeType> some_value;
> some_value.instantiate();
> some_function(some_value);
> ```

I actually don't agree. I don't think passing a new instance of `Ref<>` is actually clearer than being explicit. 

```cpp
void my_func() {
    ...
    {
        // The ideal code would scope the ref to deference it ASAP.
        Ref<SomeType> some_value;
        some_value.instantiate();
        some_function(some_value);
    }
    ...
}
```

Or, we could always make it so that `instantiate()` returns itself.

```cpp
void my_func() {
    ...
    {
        // The ideal code would scope the ref to deference it ASAP.
        Ref<SomeType> some_value;
        some_function(some_value.instantiate());
    }
    ...
}
```

**Review by akien-mga — APPROVED** [review]
Looks good.

**Pull request merged**